crypto/x509util/convert.go

```diff
@@ -0,0 +1,205 @@
+		Signature:                   cert.Signature,
+		SignatureAlgorithm:          stepx509.SignatureAlgorithm(cert.SignatureAlgorithm),
+		PublicKeyAlgorithm:          stepx509.PublicKeyAlgorithm(cert.PublicKeyAlgorithm),
+		PublicKey:                   cert.PublicKey,
+		Version:                     cert.Version,
+		SerialNumber:                cert.SerialNumber,
+		Issuer:                      cert.Issuer,
+		Subject:                     cert.Subject,
+		NotBefore:                   cert.NotBefore,
+		NotAfter:                    cert.NotAfter,
+		KeyUsage:                    stepx509.KeyUsage(cert.KeyUsage),
+		Extensions:                  cert.Extensions,
+		ExtraExtensions:             cert.ExtraExtensions,
+		UnhandledCriticalExtensions: cert.UnhandledCriticalExtensions,
+		ExtKeyUsage:                 toStepExtKeyUsage(cert.ExtKeyUsage),
+		UnknownExtKeyUsage:          cert.UnknownExtKeyUsage,
+		BasicConstraintsValid:       cert.BasicConstraintsValid,
+		IsCA:                        cert.IsCA,
+		MaxPathLen:                  cert.MaxPathLen,
+		MaxPathLenZero:              cert.MaxPathLenZero,
+		SubjectKeyId:                cert.SubjectKeyId,
+		AuthorityKeyId:              cert.AuthorityKeyId,
+		OCSPServer:                  cert.OCSPServer,
+		IssuingCertificateURL:       cert.IssuingCertificateURL,
+		DNSNames:                    cert.DNSNames,
+		EmailAddresses:              cert.EmailAddresses,
+		IPAddresses:                 cert.IPAddresses,
+		URIs:                        cert.URIs,
+		PermittedDNSDomainsCritical: cert.PermittedDNSDomainsCritical,
+		PermittedDNSDomains:         cert.PermittedDNSDomains,
+		ExcludedDNSDomains:          cert.ExcludedDNSDomains,
+		PermittedIPRanges:           cert.PermittedIPRanges,
+		ExcludedIPRanges:            cert.ExcludedIPRanges,
+		PermittedEmailAddresses:     cert.PermittedEmailAddresses,
+		ExcludedEmailAddresses:      cert.ExcludedEmailAddresses,
+		PermittedURIDomains:         cert.PermittedURIDomains,
+		ExcludedURIDomains:          cert.ExcludedURIDomains,
+		CRLDistributionPoints:       cert.CRLDistributionPoints,
+		PolicyIdentifiers:           cert.PolicyIdentifiers,
+	}
+}
+
+func toStepExtKeyUsage(eku []x509.ExtKeyUsage) []stepx509.ExtKeyUsage {
+	var ret []stepx509.ExtKeyUsage
+	for _, u := range eku {
+		ret = append(ret, stepx509.ExtKeyUsage(u))
+	}
+	return ret
+}
+
+// ToX509Certificate converts a x509.Certificate from the internal package to
+// the standard version of the x509.Certificate.
+func ToX509Certificate(cert *stepx509.Certificate) *x509.Certificate {
+	return &x509.Certificate{
+		Raw:                         cert.Raw,
+		RawTBSCertificate:           cert.RawTBSCertificate,
+		RawSubjectPublicKeyInfo:     cert.RawSubjectPublicKeyInfo,
+		RawSubject:                  cert.RawSubject,
+		RawIssuer:                   cert.RawIssuer,
+		Signature:                   cert.Signature,
+		SignatureAlgorithm:          x509.SignatureAlgorithm(cert.SignatureAlgorithm),
```

Comment:
what if the sigalg was ED25519? Won't this lead to an invalid x509.Certificate?

Reply:
Yes and no, SignatureAlgorithm is just an `int`, the problem would happen when you are signing or marshaling that certificate, not at this time.